web/firefoxcom.js

```diff
@@ -310,17 +310,67 @@ class FirefoxScripting {
 class MLManager {
   #enabled = new Map();
 
-  constructor({ enableAltText }) {
-    this.#enabled.set("altText", enableAltText);
+  constructor(options) {
+    this.eventBus = null;
+    this.enable({ ...options, listenToProgress: false });
   }
 
-  isEnabledFor(name) {
+  async isEnabledFor(name) {
     return this.#enabled.get(name);
   }
 
   guess(data) {
     return FirefoxCom.requestAsync("mlGuess", data);
   }
+
+  enable({ enableAltText, listenToProgress }) {
+    if (enableAltText) {
+      this.#loadAltTextEngine(listenToProgress);
+    }
+  }
+
+  #loadAltTextEngine(listenToProgress) {
+    if (!listenToProgress || !this.eventBus) {
+      this.#enabled.set(
+        "altText",
+        new Promise(resolve => {
+          window.addEventListener(
+            "loadAIEngineProgress",
+            ({ detail }) => {
+              resolve(detail.ok);
+            },
+            { once: true }
+          );
+          FirefoxCom.request("loadAIEngine", {
```

Comment:
*I don't know anything about the AI engine, so this might be a stupid question:*
Do we "need" to use event listeners here, and similarly below, or could we perhaps use `requestAsync` instead to return the data we need?

Reply:
Yep we could do that.
I chose to not do, to have one function `loadAIEngine` which is either called at startup if the pref `enableAltText` is true or later if the user chooses to re-enable it through the UI (see the Image settings stuff in the Figma specs).
So I thought it was overall much simpler like this.

Reply:
The patch for m-c:
https://phabricator.services.mozilla.com/D217113

Reply:
I thought about that again and you're right, I fixed.